packages/component/src/Activity/Bubble.js

```diff
@@ -18,22 +20,71 @@ const ROOT_CSS = css({
   }
 });
 
-const Bubble = ({ 'aria-hidden': ariaHidden, children, className, fromUser, nub, styleSet }) => (
-  <div
-    aria-hidden={ariaHidden}
-    className={classNames(
-      ROOT_CSS + '',
-      styleSet.bubble + '',
-      { 'from-user': fromUser, webchat__bubble_has_nub: nub },
-      className + '' || ''
-    )}
-  >
-    <div className="webchat__bubble__content">{children}</div>
-    {nub && !!(fromUser ? styleSet.options.bubbleFromUserNubSize : styleSet.options.bubbleNubSize) && (
-      <div className="webchat__bubble__nub" />
-    )}
-  </div>
-);
+function acuteNubSVG(nubSize, strokeWidth, side, upSideDown = false) {
+  const halfNubSize = nubSize / 2;
+  const halfStrokeWidth = strokeWidth / 2;
+  const horizontalTransform =
+    side === 'bot' ? '' : `translate(${halfNubSize} 0) scale(-1 1) translate(${-halfNubSize} 0)`;
+  const verticalTransform = upSideDown ? `translate(0 ${halfNubSize}) scale(1 -1) translate(0 ${-halfNubSize})` : '';
+
+  const p1 = [nubSize, halfStrokeWidth].join(' ');
+  const p2 = [strokeWidth, halfStrokeWidth].join(' ');
+  const p3 = [nubSize + strokeWidth, nubSize + halfStrokeWidth].join(' ');
+
+  return (
+    <svg
+      className="webchat__bubble__nub"
+      version="1.1"
+      viewBox={`0 0 ${nubSize} ${nubSize}`}
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <g transform={`${horizontalTransform} ${verticalTransform}`}>
+        <path d={`M${p1} L${p2} L${p3}`} />
+      </g>
+    </svg>
+  );
+}
+
+function isPositive(value) {
+  return 1 / value >= 0;
+}
+
+const Bubble = ({
+  'aria-hidden': ariaHidden,
+  children,
+  className,
+  fromUser,
+  nub,
+  styleSet: { bubble: bubbleCSS, options: styleOptions }
+}) => {
+  const {
+    bubbleBorderWidth,
+    bubbleFromUserBorderWidth,
+    bubbleFromUserNubSize,
+    bubbleNubSize,
+    bubbleNubOffset,
+    bubbleFromUserNubOffset
+  } = styleOptions;
+
+  return (
+    <div
+      aria-hidden={ariaHidden}
+      className={classNames(
+        ROOT_CSS + '',
+        bubbleCSS + '',
+        { 'from-user': fromUser, webchat__bubble_has_nub: nub },
+        className + '' || ''
+      )}
+    >
+      <div className="webchat__bubble__content">{children}</div>
+      {nub &&
+        !!(fromUser ? styleOptions.bubbleFromUserNubSize : styleOptions.bubbleNubSize) &&
```

Comment:
change variable names to include `bubbleFromBotNubSize`, etc.

Reply:
simplify code a little / leave comments?

Reply:
Simplified the code and added comments.

For `bubbleFromBot*`, created a new issue #2489. This PR is targeting a fix for Emulator and bubble nub is already in the code for quite some time.